fvm/meter/computation_meter.go

```diff
@@ -0,0 +1,121 @@
+package meter
+
+import (
+	"math"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/fvm/errors"
+)
+
+// TODO(patrick): rm after https://github.com/onflow/flow-emulator/pull/229
+// is merged and integrated.
+const (
+	ComputationKindCreateAccount = 2006
+	ComputationKindGetValue      = 2020
+	ComputationKindSetValue      = 2026
+)
+
+type MeteredComputationIntensities map[common.ComputationKind]uint
+
+var (
+	// DefaultComputationWeights is the default weights for computation intensities
+	// these weighs make the computation metering the same as it was before dynamic execution fees
+	// these weighs make the computation metering the same as it was before dynamic execution fees
+	DefaultComputationWeights = ExecutionEffortWeights{
+		common.ComputationKindStatement:          1 << MeterExecutionInternalPrecisionBytes,
+		common.ComputationKindLoop:               1 << MeterExecutionInternalPrecisionBytes,
+		common.ComputationKindFunctionInvocation: 1 << MeterExecutionInternalPrecisionBytes,
+	}
+)
+
+// MeterExecutionInternalPrecisionBytes are the amount of bytes that are used internally by the WeigthedMeter
+// to allow for metering computation smaller than one unit of computation. This allows for more fine weights.
+// A weight of 1 unit of computation is equal to 1<<16. The minimum possible weight is 1/65536.
+const MeterExecutionInternalPrecisionBytes = 16
+
+type ExecutionEffortWeights map[common.ComputationKind]uint64
+
+type ComputationMeterParameters struct {
+	computationLimit   uint64
+	computationWeights ExecutionEffortWeights
+}
+
+func DefaultComputationMeterParameters() ComputationMeterParameters {
+	// This is needed to work around golang's compiler bug
+	umax := uint(math.MaxUint)
+	return ComputationMeterParameters{
+		computationLimit:   uint64(umax) << MeterExecutionInternalPrecisionBytes,
+		computationWeights: DefaultComputationWeights,
+	}
+}
+
+func (params MeterParameters) WithComputationLimit(limit uint) MeterParameters {
+	newParams := params
+	newParams.computationLimit = uint64(limit) << MeterExecutionInternalPrecisionBytes
+	return newParams
+}
+
+func (params MeterParameters) WithComputationWeights(
+	weights ExecutionEffortWeights,
+) MeterParameters {
+	newParams := params
+	newParams.computationWeights = weights
+	return newParams
+}
+
+func (params ComputationMeterParameters) ComputationWeights() ExecutionEffortWeights {
+	return params.computationWeights
+}
+
+// TotalComputationLimit returns the total computation limit
+func (params ComputationMeterParameters) TotalComputationLimit() uint {
+	return uint(params.computationLimit >> MeterExecutionInternalPrecisionBytes)
+}
+
+type ComputationMeter struct {
+	params ComputationMeterParameters
+
+	computationUsed        uint64
+	computationIntensities MeteredComputationIntensities
+}
+
+func NewComputationMeter(params ComputationMeterParameters) *ComputationMeter { // This is needed to work around golang's compiler bug
```

Comment:
could you clarify what's the compiler bug?

Reply:
You added that line, I just moved it, I don't actually know :)
https://github.com/onflow/flow-go/commit/62033da8d142a4ed9ba98a367055f611451fb03b

Reply:
could you rm the comment.  it may not make sense any more